containers/ecr-viewer/src/app/api/services/core_database_repo.ts

```diff
@@ -0,0 +1,337 @@
+    if (criteria[criterium] !== undefined && criteria[criterium] !== null) {
+      query = query.where(criterium, "=", criteria[criterium]);
+    }
+  }
+
+  return await query.selectAll().execute();
+}
+
+/**
+ * Creates an eCR condition object
+ * @async
+ * @function createEcrCondition
+ * @param condition - the NewECRConditions to be created
+ * @returns the created eCR condition object
+ */
+export async function createEcrCondition(
+  condition: NewECRConditions | null,
+): Promise<ECRConditions | undefined> {
+  if (!condition || condition === null) {
+    throw new Error("eICR Data is required.");
+  }
+  try {
+    return await (db as Kysely<Core>)
+      .insertInto("ecr_rr_conditions")
+      .values(condition)
+      .returningAll()
+      .executeTakeFirstOrThrow();
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+/**
+ * Updates an eCR condition object
+ * @async
+ * @function updateEcrCondition
+ * @param uuid - the UUID of the eCR Condition to be updated
+ * @param updateWith - the ECRConditionsUpdate to be applied
+ * @returns the updated eCR condition object
+ */
+export async function updateEcrCondition(
+  uuid: string,
+  updateWith: ECRConditionsUpdate,
+): Promise<void> {
+  await (db as Kysely<Core>)
+    .updateTable("ecr_rr_conditions")
+    .set(updateWith)
+    .where("uuid", "=", uuid)
+    .execute();
+}
+
+/**
+ * Deletes an eCR condition object
+ * @async
+ * @function deleteEcrCondition
+ * @param uuid - thge UUID of the record to be deleted
+ * @returns the deleted eCR condition object
+ */
+export async function deleteEcrCondition(
+  uuid: string,
+): Promise<ECRConditions | undefined> {
+  const ecr = await findEcrConditionById(uuid);
+
+  if (ecr) {
+    await (db as Kysely<Core>)
+      .deleteFrom("ecr_rr_conditions")
+      .where("uuid", "=", uuid)
+      .execute();
+  }
+
+  return ecr;
+}
+
+// ECR_RR_RULE_SUMMARIES
+
+/**
+ * Finds an eCR rule summary by its ID
+ * @async
+ * @function findEcrRuleById
+ * @param id - the ID of the eCR being looked up
+ * @returns an eCR rule object
+ */
+export async function findEcrRuleById(
+  id: string,
+): Promise<ECRRuleSummaries | undefined> {
+  return await (db as Kysely<Core>)
+    .selectFrom("ecr_rr_rule_summaries")
+    .where("uuid", "=", id)
+    .selectAll()
+    .executeTakeFirst();
+}
+
+/**
+ * Finds an eCR rule summary by its criteria
+ * @async
+ * @function findEcrRule
+ * @param criteria - the Partial<ECRRuleSummaries> filtering criteria
+ * @returns an eCR rule object
+ */
+export async function findEcrRule(
+  criteria: Partial<ECRRuleSummaries>,
+): Promise<ECRRuleSummaries[]> {
+  let query = (db as Kysely<Core>).selectFrom("ecr_rr_rule_summaries");
+
+  for (const criterium of Object.keys(criteria) as (keyof ECRRuleSummaries)[]) {
+    if (criteria[criterium] !== undefined && criteria[criterium] !== null) {
+      query = query.where(criterium, "=", criteria[criterium]);
+    }
+  }
+
+  return await query.selectAll().execute();
```

Comment:
is it worth abstracting out this pattern to a helper function? it's used in quite a few spots with very minimal changes